src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -152,21 +157,45 @@ private static BidRequest fillExplicitParameters(BidRequest bidRequest, RoutingC
      * This method extracts parameters from http request and overrides corresponding attributes in {@link BidRequest}.
      */
     private BidRequest overrideParameters(BidRequest bidRequest, HttpServerRequest request) {
-        final Site updatedSite = overrideSitePage(bidRequest.getSite(), request);
+        final Site updatedSite = overrideSite(bidRequest.getSite(), request);
         final Imp updatedImp = overrideImp(bidRequest.getImp().get(0), request);
         final Long updatedTimeout = updateTimeout(request);
 
         return updateBidRequest(bidRequest, updatedSite, updatedImp, updatedTimeout);
     }
 
-    private static Site overrideSitePage(Site site, HttpServerRequest request) {
+    private static Site overrideSite(Site site, HttpServerRequest request) {
         final String canonicalUrl = canonicalUrl(request);
-        if (StringUtils.isBlank(canonicalUrl)) {
-            return site;
-        }
 
-        final Site.SiteBuilder siteBuilder = site == null ? Site.builder() : site.toBuilder();
-        return siteBuilder.page(canonicalUrl).build();
+        Site.SiteBuilder siteBuilder;
```

Comment:
I would say this logic is too verbose, overcomplicated and contains unwarranted duplications. It could be simplified as follows:
```
final JsonNode siteExt = site != null ? site.getExt() : null;
final boolean shouldSetExtAmp = siteExt == null || siteExt.get("amp") == null;

if(StringUtils.isNotBlank(canonicalUrl) || shouldSetExtAmp) {
    final Site.SiteBuilder siteBuilder = site == null ? Site.builder() : site.toBuilder();
    if(StringUtils.isNotBlank(canonicalUrl)) {
        siteBuilder.page(canonicalUrl);    
    }
    if(shouldSetExtAmp) {
        siteBuilder.ext(Json.mapper.valueToTree(ExtSite.of(1)));
    }
    
    return siteBuilder.build();
}

return null;
```